src/Agent.Sdk/Knob/AgentKnobs.cs

```diff
@@ -230,6 +230,13 @@ public class AgentKnobs
             new RuntimeKnobSource("ALLOW_TFVC_UNSHELVE_ERRORS"),
             new EnvironmentKnobSource("ALLOW_TFVC_UNSHELVE_ERRORS"),
             new BuiltInDefaultKnobSource("false"));
-    }
 
+        // Set ALLOW_JAVA_CAPABILITY_HIGHER_THAN_9 variable with any value
+        // to disable recognition of Java higher than 9
+        public static readonly Knob DisableRecognitionOfJDKHigherThen9 = new Knob(
+            nameof(DisableRecognitionOfJDKHigherThen9),
+            "Allow to recognise SDK higher than 9",
+            new EnvironmentKnobSource("DISABLE_JAVA_CAPABILITY_HIGHER_THAN_9"),
+            new BuiltInDefaultKnobSource(string.Empty));
```

Comment:
Could it make sense to set 'true' instead?

Reply:
I assume you mean [this](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/test-path?view=powershell-7.1#example-7--test-a-path-with-null-as-the-value) case from PS documentation. If so it is about the null value that is passed as _Path_ argument, in our case the value of the argument is always corresponding to string _env:DISABLE_JAVA_CAPABILITY_HIGHER_THAN_9_ . 
You can try to execute _Test-Path_ command with the not existing env variable and you will see that in this case, it will return false, not throw an exception.

Reply:
May be I just misunerstood the idea of this knob, but since the env variable is not setting by default, in my understanding this value should be null. I have also seen such approach in [TraceVerbose](https://github.com/microsoft/azure-pipelines-agent/blob/15df9f008c27abcf941a11989cdd8100a4d3952a/src/Agent.Sdk/Knob/AgentKnobs.cs#L98) knob.